static/js/ui_init.js

```diff
@@ -584,10 +584,10 @@ export function initialize_everything() {
     settings_toggle.initialize();
     about_zulip.initialize();
 
-    // All overlays must be initialized before hashchange.js
-    hashchange.initialize();
     unread_ui.initialize();
     activity.initialize();
+    // All overlays must be initialized before hashchange.js
+    hashchange.initialize(); // Must happen after activity.initialize()
```

Comment:
Hmm, I think we could instead address this issue by having `activity.redraw()` be a noop if the `activity` module hasn't been initialized yet, which is easy to keep track of with a global variable within that module.

(Since fundamentally it'll get drawn properly afterwards).

That said, can you explain the call path through which initializing the hashchange module ends up calling `activity.redraw`?  I'm not sure I understand why that would happen.

Reply:
Yeah my bad,  I didn't confirm the changes in this commit. 

Actually the reason, we are initialising `hashchange` after `activity` is because when the hash is `/`, `/#` or `#recent_topics`, it calls [`recent_topics_ui.show()`](https://github.com/zulip/zulip/blob/1bdb7b1141e575f7cd4c0b259363b2979a2da4f2/static/js/recent_topics_ui.js#L577) and cause [`recent_topics_util.is_in_focus()`](https://github.com/zulip/zulip/blob/1bdb7b1141e575f7cd4c0b259363b2979a2da4f2/static/js/recent_topics_util.js#L11) to become `true`, which will lead this [code to execute](https://github.com/zulip/zulip/pull/19541/commits/40a6017229607b6d2ebfcc9d6950730ac7026a8d#diff-0610f2177e740845f3941acfaefa8097f2c86eac9c4f7bcaacbe0a91e650ccadR515-R517) and eventually cause error in 

https://github.com/zulip/zulip/blob/1bdb7b1141e575f7cd4c0b259363b2979a2da4f2/static/js/recent_topics_ui.js#L153-L156


I have fixed this by adding a check in `recent_topics_util.is_in_focus()` to confirm at least one row is there.